master/internal/api_trials.go

```diff
@@ -32,21 +32,17 @@ func trialStatus(d *db.PgDB, trialID int32) (model.State, int, error) {
 func (a *apiServer) TrialLogs(
 	req *apiv1.TrialLogsRequest, resp apiv1.Determined_TrialLogsServer) error {
 	if err := grpc.ValidateRequest(
-		func() (bool, string) { return req.Limit >= 0, "Limit must be >= 0" },
+		grpc.ValidateLimit(req.Limit),
 	); err != nil {
 		return err
 	}
 	_, total, err := trialStatus(a.m.db, req.TrialId)
 	if err != nil {
 		return err
 	}
-	offset := int(req.Offset)
-	if req.Offset < 0 {
-		offset = total + offset
-		if offset < 0 {
-			offset = 0
-		}
-	}
+
+	offset := effectiveOffset(int(req.Offset), total)
+
 	if limit := int32(total - offset); !req.Follow && (limit < req.Limit || req.Limit == 0) {
```

Comment:
we could probably use `effectiveLimit` here :thinking:  but I rather not touch it in this PR.